src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3541,23 +3543,24 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
                   case AssignOrNamedArg(lhs@Ident(name), rhs) =>
                     // named args: only type the righthand sides ("unknown identifier" errors otherwise)
                     // the assign is untyped; that's ok because we call doTypedApply
-                    typedArg0(rhs) match {
-                      case (rhsTyped, tp) => (treeCopy.AssignOrNamedArg(arg, lhs, rhsTyped), NamedType(name, tp))
-                    }
+                    val rhsTyped = typedArg0(rhs)
+                    addArgAndType(treeCopy.AssignOrNamedArg(arg, lhs, rhsTyped), NamedType(name, rhsTyped.tpe.deconst) )
                   case treeInfo.WildcardStarArg(_) =>
-                    typedArg0(arg) match {
-                      case (argTyped, tp) => (argTyped, RepeatedType(tp))
-                    }
+                    val argTyped = typedArg0(arg)
+                    addArgAndType(argTyped, RepeatedType(argTyped.tpe.deconst) )
                   case _ =>
-                    typedArg0(arg)
+                    val argTyped = typedArg0(arg)
+                    addArgAndType(argTyped, argTyped.tpe.deconst)
                 }
-              }.unzip
+              }
             }
             if (context.reporter.hasErrors)
               setError(tree)
             else {
-              inferMethodAlternative(fun, undetparams, argTpes, pt)
-              doTypedApply(tree, adaptAfterOverloadResolution(fun, mode.forFunMode, WildcardType), args1, mode, pt)
+              inferMethodAlternative(fun, undetparams, argTpes.toList, pt)
+              val args1List = args1.toList
+              val args1Conserved = if (args == args1.toList) args else args1List
```

Comment:
What is the benefit of using the `conserved` here? The `args1` list, and its nodes, are already created by this point. Perhaps, could there be a test that may be run before the `foreach` loop, that would determine if the `args1` would be needed in the first place?

Reply:
There probably isn't much benefit. It _might_ be better to throw away the newly created list quickly and retain the existing list to better suit generational garbage collection.

Looking back at this slightly awkward code made me realise that we can simplify it further by using `map2conserve` rather than `foreach2` above, and just using the mutable buffer as for the `argTpes`. I've pushed that suggestion.